File: class2relational/src/main/java/org/atlanmod/scratch/relationalModel/RelationalLink.java
package org.atlanmod.scratch.relationalModel;

import org.atlanmod.Link;

import java.util.List;

public class RelationalLink implements Link<RelationalElement, String, List<RelationalElement>> {

    private RelationalElement source;
    private String ref;
    private List<RelationalElement> target;

    public RelationalLink(RelationalElement source, String ref, List<RelationalElement> target){
        this.source = source;
        this.ref = ref;
        this.target = target;
    }

    @Override
    public RelationalElement getSource() {
        return this.source;
    }

    @Override
    public String getReference() {
        return this.ref;
    }

    @Override
    public List<RelationalElement> getTarget() {
        return this.target;
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/relationalmodel/RelationalLink.scala
package org.atlanmod.model.relationalmodel

import org.atlanmod.model.DynamicLink
import org.atlanmod.tl.util.ListUtils

abstract class RelationalLink (type_ : String, source: RelationalElement, target: List[RelationalElement])
  extends DynamicLink(type_, source, target){

    override def equals(o: Any): Boolean = {
        o match {
            case obj: RelationalLink => obj.getType.equals(type_) & obj.getSource.equals(source) & obj.getTarget.equals(target)
            case _ => false
        }
    }

    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: RelationalLink =>
                obj.getSource.weak_equals(source) &&
                  obj.getType.equals(type_) &&
                  ListUtils.weak_eqList(obj.getTarget, target)
            case _ => false
        }
    }
}